### PR TITLE
TreeViewerHandler can retrieve more matching tree items

### PR DESCRIPTION
**Pull request by mlabuda**
At the moment tree viewer handler just get a single item by its non styled text. If there are several such items in tree, exception is thrown. There should be a method getItems to get all items matching specified path.

### CODE REVIEW

**Comment by mlabuda** [comment]
I've also cleaned up code. There were 2 methods for retrieving single tree item by single name or full path. I've made universal single method for retrieving single/more tree item(s). Tests are also revamped, tree on which tests are executed is improved. Plus some other changes.

**Comment by mlabuda** [comment]
PR rerun on Mac bcs of timeout, see http://machydra.brq.redhat.com:8080/job/RedDeer_verification_matrix/2634/

File: plugins/org.jboss.reddeer.jface/src/org/jboss/reddeer/jface/viewer/handler/TreeViewerHandler.java [hunk 1]
```diff
@@ -87,74 +56,91 @@ public TreeItem getTreeItem(Tree tree, String name) {
...
 	 */
-	public TreeItem getTreeItem(TreeItem treeItem, String... path) {
-		TreeItem item = treeItem;
-		int index = 0;
+	public List<TreeItem> getTreeItems(TreeItem item, String... path) {
+		return getTreeItems(item.getItems().iterator(), path);
+	}
+	
+	private List<TreeItem> getTreeItems(Iterator<TreeItem> iterator, String... path) {
+		if (path.length == 0 || path == null) {
```
Comment (rawagner) [inline plugins/org.jboss.reddeer.jface/src/org/jboss/reddeer/jface/viewer/handler/TreeViewerHandler.java]: null check has to be before length check

File: plugins/org.jboss.reddeer.jface/src/org/jboss/reddeer/jface/viewer/handler/TreeViewerHandler.java [hunk 2]
```diff
@@ -87,74 +56,91 @@ public TreeItem getTreeItem(Tree tree, String name) {
...
+		int index = 0;               
+		Iterator<TreeItem> nestedItemsIterator = iterator;
+		List<TreeItem> resultItems = new ArrayList<TreeItem>();
+	
+		while (index < path.length) {
+			List<TreeItem> nestedItems = new ArrayList<TreeItem>();
+			while (nestedItemsIterator.hasNext()) {
+				TreeItem item = nestedItemsIterator.next();
+				if (getNonStyledText(item).equals(path[index])) {
+					if (!item.isExpanded()) {
```
Comment (rawagner) [inline plugins/org.jboss.reddeer.jface/src/org/jboss/reddeer/jface/viewer/handler/TreeViewerHandler.java]: you need to expand item only in else statement
if index==path.length -1 then we dont need to expand tree item. Expanding item is quite lengthy operation

**Pull request merged**